java/src/jmri/managers/configurexml/DefaultSignalMastManagerXml.java

```diff
@@ -156,6 +156,23 @@ public boolean load(Element shared, Element perNode) {
             }
         }
 
+        list = shared.getChildren("matrixsignalmast");
+        if (list != null) {
+            for (int i = 0; i < list.size(); i++) {
+                Element e = list.get(i);
+                String adapterName = e.getAttribute("class").getValue();
+                log.debug("load via " + adapterName);
+                try {
+                    XmlAdapter adapter = (XmlAdapter) Class.forName(adapterName).newInstance();
+                    // and do it
+                    adapter.load(e, null);
+                } catch (Exception ex) {
+                    log.error("Exception while loading " + e.getName() + ":" + ex);
+                    ex.printStackTrace();
```

Comment:
Add the exception as a parameter to the log.error statement if you want the stack trace instead of calling ex.printStackTrace, since logging may be going to a different location than STDERR.